VideoPress Onboarding v2

pgk opened the pull request:
#### Proposed Changes

* Creates a new onboarding flow for VideoPress.
* See also pxWta-1z5-p2
* This is a culmination of many different PRs that were merged to this branch.

<img width="1066" alt="Screen Shot 2022-10-12 at 5 30 42 PM" src="https://user-images.githubusercontent.com/789137/195466344-588a09d7-6005-43db-ae6f-e1d77b3e8f98.png">

#### Testing Instructions

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

* Start the flow at http://calypso.localhost:3000/setup/intro?flow=videopress
* Walk through the onboarding, choosing a premium or better plan.
* You should end up with a Videomaker site.


Related to #

pgk committed:
Create VideoPress Onboarding v2 stepper structure. (#67454)

thedebian committed:
VideoPress site options step (WIP) (#67501)

* VideoPress site options step (WIP)

* Submit options step

pgk committed:
VideoPress Onboarding v2: Account Step (#67552)

* VideoPress Onboarding v2: Create account.

* Adjust the flow steps slightly and remove nonexistent ones.

thedebian committed:
Videopress onboarding v2: domain selection (#67596)

* Submit options step

* VideoPress onboarding, domain selection step

* Choose a domain styling

* Add video, studio, productions, com as suggested tlds

thedebian committed:
Videopress onboarding: plan selection (#67643)

* Submit options step

* VideoPress onboarding, domain selection step

* Choose a domain styling

* Add video, studio, productions, com as suggested tlds

* VideoPress site options step (WIP)

* Submit options step

* VideoPress onboarding: select a plan (WIP)

* Update how we store site title, description and domain by using onboarding store

* VideoPress onboarding: choose a plan step

* Add translation for choose a domain step

* Rebase

thedebian committed:
Videopress onboarding v2 global style (#67736)

* VideoPress onboarding theme

* VideoPress powered logo

* Update flow progress and fix svg properties in videopress logo

thedebian committed:
VideoPress Onboarding v2: Site options refined (#67790)

* Update flow progress and fix svg properties in videopress logo

* Site options refined

* Populate existing site title and description in site options onboarding

thedebian committed:
VideoPress onboarding v2 : choose a domain step refined (#67857)

* VideoPress onboarding theme

* VideoPress powered logo

* Update flow progress and fix svg properties in videopress logo

* Site options refined

* Populate existing site title and description in site options onboarding

* Allow passing the products list down to the domain search results

* Choose a domain styling WIP

* Choose a domain asides and styling

* Domain styling

* Domain step, style placeholders

thedebian committed:
Add classname to body so we can target VideoPress stepper specifically (#67894)

pgk committed:
Actually create a site on VideoPress Onboarding flow v2. (#67912)

roundhill committed:
VideoPress Onboarding V2: Intro View (#67613)

* Adding intro view styles for new VideoPress flow. Refactors
 "Jetpack powered" to be more inclusive of different "powered by" products so that we can show the V
ideoPress logo there instead of Jetpack.

* New VP logo, aligned vertically

* More updates to new design iteration.

* Seeing if playsinline helps with mobile safari autoplay.

* Fix merge conflict for showing jetpack powered badge.

* No longer need z-index hack!

* Adjustments to the triangles

* Restore previous CSS classname.

* Restoring code, unintentional change.

* Move color styles out of body so they aren`t applied globally.

* Fixing CSS after bad merge.

* remove top margin to fix centering.

Co-authored-by: Eduardo Villuendas <eduardo.villuendas@automattic.com>

delputnam commented:
Howdy!  It looks like you're wanting to limit the TLDs shown in the domain suggestions for this flow.  It would be better if we could handle this on the back end rather than including this logic in the front end.  We (Nomado) can provide a custom vendor string that you can use for this as we did for the link-in-bio and newsletter flows.

thedebian committed:
Videopress onboarding v2 : plan selection refined (#67916)

* VideoPress Onboarding V2: plan selection refined

* Mobile layout for choose a plan

jgcaruso committed:
VP Onboarding: clear onboarding site options on flow start and flow completion (#68162)

* clear onboarding site options on flow start and flow completion

* clear domain as well when starting or completing videopress flow

thedebian committed:
Update/videopress onboarding v2 site creation (#68196)

* Mobile layout for choose a plan

* Trying to do stuff with things, but it's not working, WIP

* One more step: got it to the checkout page.

* Cleanup

* Cleanup

* Less parameters = more fun

* Add domain to cart

Co-authored-by: Panos Kountanis <panosktn@gmail.com>

thedebian committed:
VideoPress onboarding v2 : steps validation (#68280)

* Mobile layout for choose a plan

* Trying to do stuff with things, but it's not working, WIP

* One more step: got it to the checkout page.

* Cleanup

* Cleanup

* Less parameters = more fun

* Add domain to cart

* Steps proxy on useStepNavigation

* Upate validation rules

Co-authored-by: Panos Kountanis <panosktn@gmail.com>

thedebian committed:
VideoPress onboarding v2 : buttons feedback (#68336)

* Mobile layout for choose a plan

* Trying to do stuff with things, but it's not working, WIP

* One more step: got it to the checkout page.

* Cleanup

* Cleanup

* Less parameters = more fun

* Add domain to cart

* Steps proxy on useStepNavigation

* Upate validation rules

* Disable form button if site title is not set

* Add button feedback for plan selection

* Update style.scss

Co-authored-by: Panos Kountanis <panosktn@gmail.com>

thedebian committed:
Update/videopress onboarding v2 processing step (#68349)

* Mobile layout for choose a plan

* Trying to do stuff with things, but it's not working, WIP

* Cleanup

* VP Onboarding: clear onboarding site options on flow start and flow completion (#68162)

* clear onboarding site options on flow start and flow completion

* clear domain as well when starting or completing videopress flow

* Upate validation rules

* Disable form button if site title is not set

* VideoPress processing step WIP

* Processing step

Co-authored-by: John Caruso <johncaruso@gmail.com>

thedebian committed:
Use plan product for checkout URL instead of periodAgnosticSlug (#68438)

roundhill committed:
Adding new launchpad flow for VideoPress! (#67692)

* Adding new launchpad flow for VideoPress!

* Updating to i2 of design.

* Moving styles to videopress.scss so all styles live in one place and can use the same color vars.

* Set text-overflow: ellipsis for long domains.

* Correcting `completing-purchase` path to use camel case.

* Adding new launchpad flow for VideoPress!

* Updating to i2 of design.

* Moving styles to videopress.scss so all styles live in one place and can use the same color vars.

* Set text-overflow: ellipsis for long domains.

* Fix redirect URI from checkout page

* Add plan to cart in order to add extra data for videomaker activation

* Fixing launchpad settings for VideoPress flow.

* Update client/landing/stepper/declarative-flow/internals/steps-repository/launchpad/task-helper.ts

Co-authored-by: John Caruso <johncaruso@gmail.com>

* Update client/landing/stepper/declarative-flow/internals/steps-repository/launchpad/task-helper.ts

Co-authored-by: John Caruso <johncaruso@gmail.com>

* Update client/landing/stepper/declarative-flow/internals/steps-repository/launchpad/translations.tsx

Co-authored-by: John Caruso <johncaruso@gmail.com>

Co-authored-by: thedebian <thedebian@users.noreply.github.com>
Co-authored-by: John Caruso <johncaruso@gmail.com>

roundhill committed:
Videomaker Onboarding: Add theme selection step (#68721)

* New styles and nearly there with functionality. Need to rebase to test fully.

* Set the site_vertical_name option so the theme choice gets set properly.

* Clear the selected design when finishing or starting over the flow.

* Added responsive styles for small screen.

* Add min height for large screens, fixes continue button from jumping.

* User better import paths.

* Changes to design based on feedback.

thedebian committed:
Add transitions to the VideoPress flow (#68913)

thedebian committed:
Fix launchpad to use actionDispatch and not immediately redirect

roundhill committed:
Adding changes from #68787 due to merge conflict mess.

roundhill commented:
We've got a few code style errors, I'll fix 👀

roundhill committed:
Code style fixes.

roundhill committed:
Handling currentUser in case of null.

roundhill committed:
Disabling typescript "any" warnings.

roundhill committed:
More eslint "any" disables for code not changed in this PR.

roundhill committed:
Style lint fixes.

roundhill commented on packages/data-stores/src/site/types.ts:
```diff
@@ -264,6 +264,7 @@ export interface Cart {
 	messages: Record< 'errors' | 'success', unknown >;
 }
 
+/* eslint-disable @typescript-eslint/no-explicit-any */
```
I don't feel so great about disabling the linter for this section. But, we didn't touch this code and it must be a new rule that's being enforced for typescript code?

thedebian committed:
Add skip action to "already own a domain"

thedebian committed:
VideoPress Onboarding intro: Display first video frame while video is loading (#69078)

roundhill commented:
- [ ] Bug: The `Upload your first video` launchpad step stopped working. The `site_intent` option is not being set to `videopress`.

roundhill committed:
Fixed issue where site options were not being set which caused the launchpad video upload step to stop working.

roundhill committed:
Unintentionally removed an import line.

roundhill commented:
> * If the viewport is wide but short, the line-height of the main heading is too tall. It may just be a breakpoint missing something; it looks perfect when it's narrower.

<img width="1256" alt="Screen Shot 2022-10-18 at 1 07 27 PM" src="s://user-images.githubusercontent.com/789137/196521766-1700b179-eaac-4f4c-bd63-3186fda91e0f.png">

This ok @evilluendas? It changes it to `36px` line height.

roundhill committed:
Addressing design feedback for intro page.

evilluendas commented:
> This ok @evilluendas? It changes it to 36px line height.

Yep, looks great!

thedebian committed:
Style adjustments

thedebian commented:
Site options ✅

thedebian committed:
Style adjustments for theme picker

thedebian committed:
Revert powered by videopress absolute position

thedebian commented:
Theme picker ✅

thedebian committed:
Header margin adjustments

thedebian committed:
Customize processing step page

thedebian commented:
Processing step ✅

thedebian committed:
Launchpad, use computer as default preview device if window width >= 1430 px

thedebian committed:
Launchpad, yellowize the thing

pgk committed:
Powered by VideoPress always at bottom.

This solution is flexbox based, the main ingredient being
that .step-container is a flexbox (column) and that content
flex-grows to fill all the space.

pgk committed:
Goto /start/videopress-account/user for account creation.

We render the old signup page on that url..

roundhill commented:
> * Can we replace the videos that are included by default so instead of the black and white street, the fish, etc. we include some of the clips used in the first step? If it's feasible, I can prepare the clips so they aren't huge and send them your way. We could also credit the authors that way, although it's not necessary by license.

Yes I just need the videos themselves, or you can upload them to a public site (Like maybe the videopress.com site) and give me the video GUIDs.

roundhill committed:
Show the progress bar on the VideoPress user account view.

pgk commented:
> I tried going back clicking on the upper-left W logo and then Dashboard, and it showed me a "what are your goals" screen. Is this deliberate? The options are kind of weird in the context of VideoPress:

@evilluendas Noting that when the launchpad checklist is incomplete, the behavior is different: W -> Dashboard  navigates to launchpad.

pgk committed:
VideoPress Launchpad: enable preview interactions.

Only when VideoPress is the flow, we allow interactions in the
launchpad preview. Useful if users want to play the homepage videos.

pgk commented:
> How feasible it is to make the videos in the preview actually playable? Would it make sense? I can imagine folks getting there and trying to play the videos and not understanding what is going on.

@evilluendas added in https://github.com/Automattic/wp-calypso/pull/67539/commits/5e757a6ea040eb3dcec79cb239bb29eca4b5ee54 . We are allowing interacting with the preview iframe when the flow is VideoPress. This also means that users can do more than play videos. I am undecided if we want this or not though.

evilluendas commented:
> We are allowing interacting with the preview iframe when the flow is VideoPress. This also means that users can do more than play videos. I am undecided if we want this or not though.

To be honest I'm not sure either — on one hand it feels like folks will immediately try to play the videos, but being able to actually browse the site may be opening the door to breaking stuff (for example if they press an external link and end up browsing a completely different site there). An alternative would be to actually take you to the editor if you click on the preview, but it should show some kind of message on hover like "Edit your homepage", but we shouldn't veer so far away from the default launchpad.

I suggest we give it a try to allowing interacting with the preview, and revert that change if we feel it's just too much.

roundhill commented:
> * When clicking "Save", would it make sense to take you back automatically to the launchpad? Or perhaps show a modal window celebrating your changes and a CTA to get back? I suspect it's super easy to get stuck there and not know how to continue.

I started looking into this and it was tough to get a redirect to work because we are living in an iframe when we're in the site editor, so there's lots of cross-origin issues. One thing that does work now is that as soon as you try to go _anywhere else_ after saving the editor (Such as clicking on the `W` and the `Dashboard`), you're taken back to the launchpad 👍 So that step where you saw "what are your goals" isn't there now and instead you see the launchpad.

roundhill committed:
Fixed stacked theme buttons when screen was medium to large size.

thedebian committed:
Substract 10px to 100vh to compute the stepper min height

roundhill committed:
Video Onboarding: Add video upload celebration modal 🎉 (#69394)

* Initial kind-of-working redirect after saving site with videopress intent.

* Adding check that redirect has occurred. Disabled for atomic sites.

* Removing deleted store item.

* Deleting stylesheet, unused.

* Remove style import, set redirect to wp.com host.

* Adding a hopefully helpful comment about the comonents goal.

* Changed from using a redirect to showing a celebration modal.

* Use window location to get the site slug

* Fix component naming and removed unrelated commits.

* Linting fixes... not sure why as did not touch this code.

jgcaruso commented on packages/data-stores/src/onboard/actions.ts:
```diff
@@ -31,16 +31,68 @@ export const setPatternContent = ( patternContent: string ) => ( {
+			lang_id: lang_id,
+			site_creation_flow: 'videopress',
+			enable_fse: true,
+			theme: defaultTheme,
```
```suggestion
			theme: 'pub/twentytwentytwo', // NOTE: keep this a consistent, free theme so post ids during headstart re-run after premium theme switch remain consistent,
```

since VideoMaker is a premium theme, it seems like the Create Site code is selecting a random theme to use for site creation. We need to set this to a consistent free theme so that the post ids are predictable so the homepage links can be properly created.

`site_vertical` is being used currently to set the theme after purchase, so this change shouldn't have any negative effects.

roundhill committed:
Fix errors introduced from upstream.

roundhill committed:
Update packages/data-stores/src/onboard/actions.ts

Co-authored-by: John Caruso <johncaruso@gmail.com>

roundhill committed:
Removing unused i18n import.

roundhill committed:
Merge branch 'add/videopress-onboarding-v2' of github.com:Automattic/wp-calypso into add/videopress-onboarding-v2

mikeyarce commented on client/landing/stepper/declarative-flow/internals/videopress.scss:
```diff
@@ -0,0 +1,237 @@
+body.is-videopress-stepper {
+	background-color: #010101;
+	color: $videopress-theme-base-text-color;
+
```
```suggestion

	.search-filters__checkbox-label{
		color: #000;
	}
```

While looking at this flow I noticed that the checkbox in the filter is currently not visible.  This will make sure it's visible:
<img width="425" alt="Screenshot 2022-11-01 at 4 21 03 PM" src="https://user-images.githubusercontent.com/3220162/199360350-e3ea4441-ec01-4008-89cc-dd985c92856e.png">

roundhill committed:
Adding config for VideoPress flow. Enabled for dev, horizon, staging and wpcalypso.

roundhill committed:
Fixing styles for `is-primary` button.

roundhill committed:
Update client/landing/stepper/declarative-flow/internals/videopress.scss

Co-authored-by: Mikey Arce <mikeyarce@gmail.com>

roundhill committed:
Merge branch "trunk" into add/videopress-onboarding-v2

# Conflicts:
#	client/landing/stepper/declarative-flow/internals/steps-repository/intro/index.tsx
#	client/landing/stepper/declarative-flow/internals/steps-repository/intro/styles.scss
#	client/landing/stepper/declarative-flow/internals/steps-repository/launchpad/task-helper.ts
#	client/landing/stepper/index.tsx

roundhill committed:
Style lint fix.

roundhill committed:
Add type cast to help with tests.

roundhill committed:
Fix redirect urls to use new path structure instead of using the `flow` parameter.

roundhill committed:
Style the new email activation banner to fit with the VideoPress onboarding theme.

roundhill committed:
Merge branch trunk into add/videopress-onboarding-v2

# Conflicts:
#	client/landing/stepper/declarative-flow/internals/steps-repository/intro/index.tsx
#	client/landing/stepper/declarative-flow/internals/steps-repository/intro/styles.scss
#	client/landing/stepper/declarative-flow/internals/steps-repository/launchpad/task-helper.ts
#	client/landing/stepper/index.tsx

roundhill reviewed (approved):
We're ready!

a8ci18n commented:
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/7619089

Thank you @pgk for including a screenshot in the description! This is really helpful for our translators.

a8ci18n commented:
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/7619089).

a8ci18n commented:
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/7619089).
